transport/ice/IceSession.cpp

```diff
@@ -984,22 +1050,50 @@ void IceSession::stateCheck(const uint64_t now)
     }
     else if (_state == State::CONNECTED)
     {
-        freezePendingProbes(now);
+        if (hasNomination())
+        {
+            freezePendingProbes(now);
+        }
+        else
+        {
+            nominate(now);
+            if (hasNomination())
+            {
+                if (_eventSink)
+                {
+                    _eventSink->onIceCandidateChanged(this,
+                        _nomination->localEndpoint.endpoint,
+                        _nomination->remoteCandidate.address);
+                }
+                freezePendingProbes(now);
+            }
+        }
     }
 }
 
 void IceSession::freezePendingProbes(uint64_t now)
 {
     for (auto& candidatePair : _candidatePairs)
     {
-        if (!candidatePair->nominated && candidatePair->state < CandidatePair::State::Failed &&
-            candidatePair->localCandidate.transportType == ice::TransportType::TCP)
+        if (!candidatePair->nominated && candidatePair->state <= CandidatePair::State::Succeeded)
```

Comment:
Why did you remove the `candidatePair->localCandidate.transportType == ice::TransportType::TCP` condition?

I think probes are like a keep-alive, if we don't probe them, NAT gateways can close them potentially by inactivity and then if we need to re-probe they might fail.